src/azure-cli/azure/cli/command_modules/storage/operations/account.py

```diff
@@ -476,6 +499,49 @@ def update_storage_account(cmd, instance, sku=None, tags=None, custom_domain=Non
     if allow_cross_tenant_replication is not None:
         params.allow_cross_tenant_replication = allow_cross_tenant_replication
 
+    if any([immutability_period_since_creation_in_days, immutability_policy_state, allow_protected_append_writes is not None]):
+        ImmutableStorageAccount = cmd.get_models('ImmutableStorageAccount')
+        AccountImmutabilityPolicyProperties = cmd.get_models('AccountImmutabilityPolicyProperties')
+        immutability_policy = None
+        from azure.cli.core.azclierror import InvalidArgumentValueError
+
+        if not instance.immutable_storage_with_versioning.enabled:
+            raise InvalidArgumentValueError("Incorrect usage: The account level immutability has been disabled. "
+                                            "The property is immutable and can only be set to true at the account "
+                                            "creation time.")
+
+        existing_policy = instance.immutable_storage_with_versioning.immutability_policy
+
+        if not existing_policy:
+            if not all([immutability_period_since_creation_in_days, immutability_policy_state, allow_protected_append_writes is not None]):
+                raise InvalidArgumentValueError("Incorrect usage: To create the account level immutability policy, "
+                                                "need to specify all of the following arguments:"
+                                                "--immutability-period --immutability-state "
+                                                "--allow-append")
+        else:
+            if existing_policy.state == 'Disabled':
+                if immutability_policy_state and immutability_policy_state == 'Locked':
+                    raise InvalidArgumentValueError(
+                        "Incorrect usage: The immutability policy state can only be changed to Locked from Unlocked.")
+            if existing_policy.state == 'Locked':
+                if immutability_policy_state and immutability_policy_state != 'Locked':
+                    raise InvalidArgumentValueError("Incorrect usage: The immutability policy state cannot be changed once Locked.")
```

Comment:
Please leave the checks to service side. We just need to pass what user give to server.

Reply:
Got it, catching HTTPResponseErrors in the test as well.